standalone/pruntime/enclave/src/contracts/btc_lottery.rs

```diff
@@ -0,0 +1,359 @@
+                        utxo: self.utxo.get(&round_id).unwrap().clone(),
+                    })
+                }
+            }
+        };
+        match inner() {
+            Err(error) => Response::Error(error),
+            Ok(resp) => resp,
+        }
+    }
+
+    fn handle_event(&mut self, ce: chain::Event) {
+        if let chain::Event::pallet_kitties(pe) = ce {
+            if let chain::pallet_kitties::RawEvent::NewLottery(total_count, winner_count) = pe {
+                let raw_seed = LOTTERY.as_bytes();
+                let mut nonce = 1;
+                // This indicates the token's kind, let's just suppose that 1 represents box,
+                // maybe if we introduce more tokens later, this can be formulated strictly
+                let kind = 1;
+                // For now, the token id is 0....01000...000, this will be used to construct the
+                // box ID below, combined with random index id and NFT flags
+                let token_id: U256 = U256::from(kind) << 128;
+                for token in 0..total_count {
+                    let mut rng = rand::thread_rng();
+                    let seed: [u8; 16] = rng.gen();
+                    let raw_data = (seed, nonce);
+                    nonce += 1;
+                    let hash_data = blake2_128(&Encode::encode(&raw_data));
+                    let index_id = u128::from_be_bytes(hash_data);
+                    let nft_id = (token_id + index_id) | *TYPE_NF_BIT;
+                    let token_id = format!("{:#x}", nft_id);
+                    let mut round_token = match self.token_set.get(&self.round_id) {
+                        Some(_) => self.token_set.get(&self.round_id).unwrap().clone(),
+                        None => Vec::new(),
+                    };
+                    round_token.push(token_id);
+                    self.token_set.insert(self.round_id, round_token);
+                }
+                let mut r = sgx_rand::thread_rng();
+                let sample = sample(
+                    &mut r,
+                    self.token_set.get(&self.round_id).unwrap().iter(),
+                    winner_count as usize,
+                );
+                let mut lottery_token = match self.lottery_set.get(&self.round_id) {
+                    Some(_) => self.lottery_set.get(&self.round_id).unwrap().clone(),
+                    None => BTreeMap::<String, PrivateKey>::new(),
+                };
+                for winner_id in sample {
+                    let mut salt = self.round_id * 10000;
+                    let s = Secp256k1::new();
+                    let raw_data = (raw_seed, salt);
+                    let seed = blake2_128(&Encode::encode(&raw_data));
+                    let pk = ExtendedPrivKey::new_master(Network::Bitcoin, &seed)
+                        .unwrap()
+                        .private_key;
+                    lottery_token.insert(String::from(winner_id), pk);
+                    salt += 1;
```

Comment:
The logic was wrong, because I put the declaration of `salt` inside the loop which was totally useless. I changed this logic, now the salt's function is normal.